### PR TITLE
Wdb 87-88/setup forgot password endpoint

### PR DESCRIPTION
**Pull request by kibartas**
My greatest condolences, I did two tasks in this one PR.

### CODE REVIEW

File: backend/backend/Controllers/UsersController.cs [hunk 1]
```diff
@@ -130,5 +131,73 @@ public ActionResult Verify(Guid id)
...
+        {
+            if (string.IsNullOrWhiteSpace(email))
+            {
+                return BadRequest();
+            }
+
+            var user = _db.Users.FirstOrDefault(x => x.Email == email);
+            if (user == null)
+            {
+                return Ok();
```
Comment (tmaciulis22) [inline backend/backend/Controllers/UsersController.cs:145]: shouldn't we return 404 here?

**Review by tmaciulis22 — CHANGES_REQUESTED** [review]

File: backend/backend/Controllers/UsersController.cs (hunk 1 above)
Comment (AArlauskas) [reply backend/backend/Controllers/UsersController.cs:145]: I believe that this was done for security reasons. If you return anything else than 200 an Evil Guy from the Dark Web would just be able to know if a person has registered in our system just by checking the response code. Or maybe Mr. @kibartas  had something else in his mind

File: backend/backend/JwtAuthentication/JwtAuthentication.cs [hunk 2]
```diff
@@ -44,5 +44,31 @@ public string Authenticate(string email, string password)
...
+                return null;
+            }
+
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var key = Encoding.ASCII.GetBytes(_tokenKey);
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(new[]
+                {
+                    new Claim("ResetPassword", "True"),
```
Comment (AArlauskas) [inline backend/backend/JwtAuthentication/JwtAuthentication.cs:61]: Could you explain the purpose of this one?

**Review by salmonela7 — APPROVED** [review]
LGTM

**Review by povilassta — APPROVED** [review]

**Pull request merged**